src/pydiode/send.py

```diff
@@ -38,24 +37,15 @@ async def read(self):
         :returns: Bytes read from STDIN
         """
         if self.regular_file:
-            # If connected to a regular file, there's no need for a
-            # StreamReader: we can quickly read the maximum amount of data for
-            # a chunk.
+            # If STDIN is a regular file, read a fixed chunk of data using a background thread.
```

Comment:
I wonder if we could used `run_in_exector()` in the other branch as well:
https://docs.python.org/3/library/asyncio-dev.html#concurrency-and-multithreading